### PR TITLE
[xsd package] added patch to fix build error caused by missing #include <iostream>

### PR DESCRIPTION
**Pull request by mdorier**
With gcc 10.3.0 the xsd package doesn't build (and probably with other compilers as well). The fix is a simple missing `#include <iostream>` in some file, which this patch adds.

### CODE REVIEW

File: var/spack/repos/builtin/packages/xsd/package.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Xsd(MakefilePackage):
     depends_on('xerces-c')
     depends_on('libtool', type='build')
 
+    patch('fix_include_iostream.patch')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/xsd/package.py]: Can you add a simple comment on top of the directive, like:
```suggestion
    # Adds a missing include for <iostream> in a source file
    patch('fix_include_iostream.patch')
```
?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request. Bonus if you can retrieve the patch from remote instead of storing it in the builtin repository (but not requested).

File: var/spack/repos/builtin/packages/xsd/package.py (hunk 1 above)
Comment (mdorier) [reply var/spack/repos/builtin/packages/xsd/package.py]: I had to adapt the patch from [here](https://git.codesynthesis.com/cgit/libxsd-frontend/libxsd-frontend/commit/?id=5029f8665190879285787a9dcdaf5f997cadd2e2), the original version has paths expressed relative to a subdirectory of the source (when the source is the tarball, which apparently contains what corresponds to multiple git repositories). If you can tell me how to specify both the URL of the patch and the fact that it should be applied from a subdirectory, I'm happy to update the PR.

Comment (alalazo) [reply var/spack/repos/builtin/packages/xsd/package.py]: I think using this:
```suggestion
    patch(
        'https://git.codesynthesis.com/cgit/libxsd-frontend/libxsd-frontend/patch/?id=5029f8665190879285787a9dcdaf5f997cadd2e2',
        sha256='d57e0aed8784d2b947983209b6513c81ac593c9936c3d7b809b4cd60d4c28607',
        working_dir='libxsd-frontend'
    )
```
and deleting the patch should work (or at least it works locally for me).

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by mdorier** [comment]
@alalazo Thanks. I see you assigned the PR to yourself, do you plan to submit the change or do you want me to do it?

**Comment by alalazo** [comment]
> Thanks. I see you assigned the PR to yourself, do you plan to submit the change or do you want me to do it?

Please go ahead with the change. We (Spack maintainers) started assigning issues / PRs to ourselves to have a better / quicker understanding of which ones got attention. Basically if we see the avatar of another maintainer we know somebody is on it :slightly_smiling_face:

**Review by alalazo — APPROVED** [review]

**Pull request merged**